### DISCUSSION TITLE
Display changing numbers or strings on a background image

### DISCUSSION THREAD

**Discussion by mohitbhoite** [Bodmer/TFT_eSPI#1886] [Q&A - General] [answered]
I love this library!

I'm attempting to display numbers and text on an image. By not passing a background color argument, I'm able to display static numbers and strings just fine, but I'm unable to figure out a way to display changing values of numbers. I'm sure there is a way to accompolish this (maybe with sprites?). Can anyone provide some pointers or the correct example to look into?

I see that this discussion [recomments using padding](https://github.com/Bodmer/TFT_eSPI/issues/986#issuecomment-774524302), but it still assumes that a background color is provided. I wish to accompolish this with a transparent background.

[edit]
I'm guessing there is no real way to accompolish this without being able to read that region of the display and rewriting it back to it. I'm using an ST7789 240x240 IPS TFT module with no data out exposed. 

Thanks!

**Comment by Bodmer** [comment, marked as answer]
It is possible using sprites but is a rather complicated process. Essentially you create 2 sprites, sprite 1 has a copy of the background area the text occupies, sprite 2 is the same size.
The background is drawn in sprite 1 with the necessary offsets to capture the correct area. When the number is to be drawn, sprite 1 is copied to sprite 2 and the number drawn in sprite 2, sprite 2 is then copied to the screen with the correct offsets.
